tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/CI/SeleniumTests.cs

```diff
@@ -0,0 +1,163 @@
+                                <h1>Test page</h1>
+                            </body>
+                        </html>
+                        """)
+                    {
+                        ContentType = "text/html"
+                    };
+                    return;
+                }
+
+                if (e.Value.PathAndQuery.EndsWith("api/v2/libraries/tests/services/setting"))
+                {
+                    e.Value.Response = new MockTracerResponse("""{"data":{"id":"b5a855bffe6c0b2ae5d150fb6ad674363464c816","type":"ci_app_tracers_test_service_settings","attributes":{"code_coverage":false,"efd_enabled":false,"flaky_test_retries_enabled":false,"itr_enabled":true,"require_git":false,"tests_skipping":true}}} """);
+                    return;
+                }
+
+                if (e.Value.PathAndQuery.EndsWith("api/v2/ci/tests/skippable"))
+                {
+                    e.Value.Response = new MockTracerResponse($"{{\"data\":[],\"meta\":{{\"correlation_id\":\"{correlationId}\"}}}}");
+                    return;
+                }
+
+                if (e.Value.PathAndQuery.EndsWith("api/v2/citestcycle"))
+                {
+                    e.Value.Headers["Content-Encoding"].Should().Be("gzip");
+
+                    var payload = JsonConvert.DeserializeObject<MockCIVisibilityProtocol>(e.Value.BodyInJson);
+                    if (payload.Events?.Length > 0)
+                    {
+                        foreach (var @event in payload.Events)
+                        {
+                            if (@event.Content.ToString() is { } eventContent)
+                            {
+                                if (@event.Type == SpanTypes.Test)
+                                {
+                                    tests.Add(JsonConvert.DeserializeObject<MockCIVisibilityTest>(eventContent));
+                                }
+                                else if (@event.Type == SpanTypes.TestSuite)
+                                {
+                                    testSuites.Add(JsonConvert.DeserializeObject<MockCIVisibilityTestSuite>(eventContent));
+                                }
+                                else if (@event.Type == SpanTypes.TestModule)
+                                {
+                                    testModules.Add(JsonConvert.DeserializeObject<MockCIVisibilityTestModule>(eventContent));
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            SetEnvironmentVariable("SAMPLES_SELENIUM_TEST_URL", $"http://127.0.0.1:{agent.Port}/evp_proxy/v4/rumpage");
+            using var processResult = await RunDotnetTestSampleAndWaitForExit(agent, packageVersion: packageVersion);
+
+            // Check if we have the data
+            tests.Count.Should().Be(1);
+            testSuites.Count.Should().Be(1);
+            testModules.Count.Should().Be(1);
```

Comment:
Don't use `.Count` in general, you get better error descriptions if you use `HasCount()`. Also, if one of these fails, you probably want to know what they _all_ have?
```suggestion
            using var s = new AssertionScope();
            tests.Should().HaveCount(1);
            testSuites.Should().HaveCount(1);
            testModules.Should().HaveCount(1);
```

Alternatively, just don't include these. Seeing as you're going to write the spans as snapshots anyway, which implicitly tests this, no?

Reply:
changed in https://github.com/DataDog/dd-go/pull/132491/files